diff --git a/dist/module/lib/step.js b/dist/module/lib/step.js
index 4df50c29f426c26bd555160101a6cdc606a7a37e..377c7e44f3a9e3cc85db5d519ddb3e6b3ed99f6d 100644
--- a/dist/module/lib/step.js
+++ b/dist/module/lib/step.js
@@ -73,6 +73,9 @@ const stepType = {
             }
             // check if there is a oneOf selection, which must be resolved
             if (targetSchema && Array.isArray(targetSchema.oneOf)) {
+                if(data === undefined || data[key] === undefined) {
+                    return targetSchema;
+                }
                 // @special case: this is a mix of a schema and optional definitions
                 // we resolve the schema here and add the original schema to `oneOfSchema`
                 return draft.resolveOneOf(data[key], targetSchema, `${pointer}/${key}`);
diff --git a/lib/step.ts b/lib/step.ts
index 267029c811874c10c8f707640e0601582d73bdbd..e5d8caebb4504d944cd9f4480b4a248604c14e5b 100644
--- a/lib/step.ts
+++ b/lib/step.ts
@@ -101,6 +101,9 @@ const stepType: Record<string, StepFunction> = {
 
             // check if there is a oneOf selection, which must be resolved
             if (targetSchema && Array.isArray(targetSchema.oneOf)) {
+                if(data === undefined || data[key] === undefined) {
+                    return targetSchema;
+                }
                 // @special case: this is a mix of a schema and optional definitions
                 // we resolve the schema here and add the original schema to `oneOfSchema`
                 return draft.resolveOneOf(data[key], targetSchema, `${pointer}/${key}`);
diff --git a/package.json b/package.json
index 0eb36ace1eefac10cc75fda876df21c91ccb8861..1d823f4e36c8b87924ccdb31597ae656fc74646c 100644
--- a/package.json
+++ b/package.json
@@ -4,7 +4,6 @@
     "description": "Customizable and hackable json-validator and json-schema utilities for traversal, data generation and validation",
     "module": "dist/module/index.js",
     "types": "dist/index.d.ts",
-    "main": "dist/jsonSchemaLibrary.js",
     "scripts": {
         "dist": "rm -rf dist; yarn dist:es5 && yarn dist:es6",
         "dist:es5": "NODE_ENV=production webpack",